python/pyspark/sql/functions.py

```diff
@@ -315,6 +312,46 @@ def _options_to_str(options):
 del _name, _doc
 
 
+class column_function(object):
+    """
+    Returns a :class:`Column` based on the given column name.
+
+    >>> __name__("firstName")
+    Column<firstName>
+
+    >>> __name__["firstName"]
+    Column<firstName>
+
+    >>> __name__.firstName
```

Comment:
Hm, I am not sure. This way will disallow other column names like white spates in the middle or special characters. Also, those expressions are already possible via `df.column_name`.

Reply:
Agree, I think this is already effectively possible.

Reply:
df.column_name is verbose if the DF has a meaningful name like customersDF:
customersDF.select(customersDF.firstName, customersDF.lastName, customersDF.zip)

The idea here is to get something as concise as the scala $.
customersDF.select(C.firstName, C.lastName, C.zip)

For whitespace, we'd need to use the full version:
customersDF.select(C("First Name"), C("Last Name), C("Zip"))

Reply:
We could:

```python
C = customersDF
customersDF.select(C.firstName, C.lastName, C.zip)
```

or

```python
C = functions.col
customersDF.select(C("First Name"), C("Last Name), C("Zip"))
```

I think we don't necessarily have to add this support - while it can be convenient to have such aliases, this can be done in user app side easily as above.